cider/function/string/StringOps.cpp

```diff
@@ -78,6 +78,24 @@ std::regex StringOp::generateRegex(const std::string& op_name,
   }
 }
 
+NullableStrType TryStringCast::operator()(const std::string& str) const {
+  CHECK("Invalid string output for TryStringCast");
+  return NullableStrType();
+}
+
+Datum TryStringCast::numericEval(const std::string_view str) const {
+  if (str.empty()) {
+    return NullDatum(return_ti_);
+  }
+  // Need to make copy for now b/c StringToDatum can mod SQLTypeInfo arg
+  SQLTypeInfo return_ti(return_ti_);
+  try {
+    return StringToDatum(str, return_ti);
```

Comment:
can we skip cast_type branch check inside StringToDatum method? eg.separate and call different cast function from here string_to_int, string_to_date...